src/mongocxx/hint.hpp

```diff
@@ -0,0 +1,76 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <mongocxx/config/prelude.hpp>
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/builder/stream/helpers.hpp>
+#include <bsoncxx/builder/stream/key_context.hpp>
+#include <bsoncxx/stdx/optional.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+using namespace bsoncxx::builder::stream;
+
+///
+/// Class representing a hint to be passed to a database operation.
+///
+class MONGOCXX_API hint {
+   public:
+    ///
+    /// Constructs a new hint.
+    ///
+    /// @param index
+    ///   Document representing the index to be used.
+    ///
+    explicit hint(bsoncxx::document::view index);
+
+    ///
+    /// Constructs a new hint.
+    ///
+    /// @param index
+    ///   String representing the name of the index to be used.
+    ///
+    explicit hint(std::string index);
+
+    template <typename T>
+    friend key_context<T> operator<<(key_context<T> builder, const hint& obj);
+
+   private:
+    bsoncxx::stdx::optional<bsoncxx::document::view> index_doc;
```

Comment:
Actually @Machyne, would you mind filing a CXX enhancement ticket to consistently use stdx names in the mongocxx directory via the mongocxx/stdx header and namespace, rather than bsoncxx?